### PR TITLE
324 breadcrumbs

### PR DESCRIPTION
**Pull request by kenjinp**
# NEW
- breadcrumbs, fully configurable according to your routes file!
- passes state to breadcrumbLabel components
- breadcrumbLabels live under /breadcrumbs in views, so you can add your own special breadcrumb component to match your route!

## Minor updates
- added inline styling to header component for reducing headaches in dev mode
- refactored reducers to add orgsFetching, projectsFetching, and ResourcesFetching flags

### CODE REVIEW

File: src/shared/components/Breadcrumb/Breadcrumb.test.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React = require('react');
+import { mount } from 'enzyme';
+import getBreadcrumbs from './getBreadcrumbs';
+import { RouteWithData } from '../../routes';
+const routes: RouteWithData[] = [
```
Comment (mdupont) [inline src/shared/components/Breadcrumb/Breadcrumb.test.tsx]: I have the feeling we'll end up having "real" routes vastly diverging from this set of test routes. How could we have 1 set of routes for which we override the component in order to do tests?

File: src/shared/components/Breadcrumb/getBreadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  const matches: Breadcrumb[] = [];
+
+  // First need to check if there is a default path for the "home page"
+  const homeMatch = maybeMatchRouteBreadcrumbWithPath(routes, '/', state);
+  if (!!homeMatch) {
+    matches.push(homeMatch as Breadcrumb);
+  }
+
+  // Now that we've found the home breadcrumb, we can continue using this method
+  // to iterate through the pathsections as defiend as between /'s
```
Comment (mdupont) [inline src/shared/components/Breadcrumb/getBreadcrumbs.tsx]: s/defiend/defined

File: src/shared/components/Breadcrumb/getBreadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  pathname
+    .replace(/\/$/, '')
```
Comment (mdupont) [inline src/shared/components/Breadcrumb/getBreadcrumbs.tsx]: To handle the home path use case:
`.replace(/([^\/])\/$/, '$1')`

File: src/shared/components/Breadcrumb/Breadcrumb.test.tsx (hunk 1 above)
Comment (kenjinp) [reply src/shared/components/Breadcrumb/Breadcrumb.test.tsx]: We can iterate with more rigorous test cases later, right now these are simple examples that test the basic use cases

File: src/shared/components/Breadcrumb/getBreadcrumbs.tsx (hunk 3 above)
Comment (kenjinp) [reply src/shared/components/Breadcrumb/getBreadcrumbs.tsx]: doesn't pass the tests :(

**Pull request merged**